test/locals_test.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# This test is going to use the RubyVM::InstructionSequence class to compile
+# local tables and compare against them to ensure we have the same locals in the
+# same order. This is important to guarantee that we compile indices correctly
+# on CRuby (in terms of compatibility).
+#
+# There have also been changes made in other versions of Ruby, so we only want
+# to test on the most recent versions.
+return if !defined?(RubyVM::InstructionSequence) || RUBY_VERSION < "3.2"
+
+require "yarp_test_helper"
+
+class LocalsTest < Test::Unit::TestCase
+  invalid = []
+  todos = []
+
+  # Invalid break
+  invalid << "break.txt"
+  invalid << "if.txt"
+  invalid << "rescue.txt"
+  invalid << "seattlerb/block_break.txt"
+  invalid << "unless.txt"
+  invalid << "whitequark/break.txt"
+  invalid << "whitequark/break_block.txt"
+
+  # Invalid next
+  invalid << "next.txt"
+  invalid << "seattlerb/block_next.txt"
+  invalid << "unparser/corpus/literal/control.txt"
+  invalid << "whitequark/next.txt"
+  invalid << "whitequark/next_block.txt"
+
+  # Invalid redo
+  invalid << "keywords.txt"
+  invalid << "whitequark/redo.txt"
+
+  # Invalid retry
+  invalid << "whitequark/retry.txt"
+
+  # Invalid yield
+  invalid << "seattlerb/dasgn_icky2.txt"
+  invalid << "seattlerb/yield_arg.txt"
+  invalid << "seattlerb/yield_call_assocs.txt"
+  invalid << "seattlerb/yield_empty_parens.txt"
+  invalid << "unparser/corpus/literal/yield.txt"
+  invalid << "whitequark/args_assocs.txt"
+  invalid << "whitequark/args_assocs_legacy.txt"
+  invalid << "whitequark/yield.txt"
+  invalid << "yield.txt"
+
+  # Dead code eliminated
```

Comment:
What does this mean? YARP gets rid of some code that CRuby doesn't?

Reply:
Opposite. If you have `false ? 1 : 2` CRuby will drop the truthy path but YARP won't.